### PR TITLE
Rebuild relation models

### PR DESCRIPTION
**Pull request by dwilliamson67**

### CODE REVIEW

File: src/Components/ComponentRebuildRelationModel.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        /// <summary>
+        /// The component relation rebuild ignored
+        /// the component should be rebuilt manually.
+        /// </summary>
+        Ignored
+    }
+
+    /// <summary>
+    /// This class represents a component that has been rebuilt by an extract service.
```
Comment (deevezina) [inline src/Components/ComponentRebuildRelationModel.cs:35]: relationship rebuild service

File: src/Components/ComponentRebuildRelationStateModel.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// Copyright (c) GlobalLink Vasont. All rights reserved.
+// </copyright>
+//-------------------------------------------------------------
+namespace Vasont.Inspire.Models.Components
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// This class contains the export state for a rebuild relation request worker within the application.
```
Comment (deevezina) [inline src/Components/ComponentRebuildRelationStateModel.cs:12]: ?

**Review by deevezina — CHANGES_REQUESTED** [review]

File: src/Components/ComponentRebuildRelationModel.cs (hunk 1 above)
Comment (dwilliamson67) [reply src/Components/ComponentRebuildRelationModel.cs:35]: fixed

File: src/Components/ComponentRebuildRelationStateModel.cs (hunk 2 above)
Comment (dwilliamson67) [reply src/Components/ComponentRebuildRelationStateModel.cs:12]: fixed

File: src/Components/ComponentRebuildRelationStateModel.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        /// Gets or sets a list of components within the database that are to be rebuilt.
+        /// </summary>
+        public List<ComponentRebuildRelationModel> Components { get; set; } = new List<ComponentRebuildRelationModel>();
+
+        /// <summary>
+        /// Gets or sets a list of current components to rebuild.
+        /// </summary>
+        public List<long> ComponentIds { get; set; } = new List<long>();
+    }
+}
```
Comment (deevezina) [inline src/Components/ComponentRebuildRelationStateModel.cs]: warning symbol here as well

File: src/Components/ComponentRebuildRelationModel.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        /// </summary>
+        Success,
+
+        /// <summary>
+        /// The component relation rebuild failed.
+        /// </summary>
+        Failed,
+
+        /// <summary>
+        /// The component relation rebuild ignored
```
Comment (deevezina) [inline src/Components/ComponentRebuildRelationModel.cs:28]: Add period after ignored
And delete line 29

**Review by deevezina — CHANGES_REQUESTED** [review]

**Review by deevezina — APPROVED** [review]

**Pull request merged**